### ISSUE TITLE
Refresh Token 

### ISSUE THREAD

**Issue by eugeniuLivandovschi** [IdentityServer/IdentityServer4#4526]
Hi,
Is there a way to achieve following flow using IdentityServer4:
 If a client is not active for a specified amount of time - he should not be able to refresh his access token. As I understand this should be achievable via RefreshTokenExpiration settings, but I don't understand what specific combination of settings should allow this behavior.None of the combinations that I tried for parameters described here: https://identityserver4.readthedocs.io/en/latest/topics/refresh_tokens.html   helped me achieve my goal

**Comment by leastprivilege** [comment]
Yes - that's the sliding refresh token lifetime...

**Comment by eugeniuLivandovschi** [comment]
OK, I am a little confused, hope you can help me
Suppose I have following settings:
AccessTokeLifetime - 30s
SlidingRefreshTokenLifetime - 60s
AbsoluteRefreshTokenLifetime - 3600s

AccesToken was created at time T1
What I expect: If I try to refresh the token at time T2 = T1 + 31s - everything should be OK, and
If I try to refresh the token an time T3 = T1 + 61s (without refreshing it in between) - it should fail
What I actually get is that at T3 the refresh passes successfully.
Please, help me understand what I am missing.
Thanks

**Comment by leastprivilege** [comment]
Have you set 

```
RefreshTokenExpiration = TokenExpiration.Sliding
```

**Comment by eugeniuLivandovschi** [comment]
OK, I found the issue. Indeed, I have not set the RefreshTokenExpiration to TokenExpiration.Sliding.
The confusion was caused by the fact that in documentation - https://identityserver4.readthedocs.io/en/latest/topics/refresh_tokens.html, 'Absolute' is described first, so I was expecting that in database  I have to set for Absolute - 0, and for Sliding - 1. Now I checked the repository and I found this mistake I was making.
Thank you.
